elf/mold.h

```diff
@@ -329,6 +329,13 @@ class InputSection {
   std::atomic_bool is_visited = false;
 
   // For ICF
+  //
+  // `leader` is the section that this section has been merged with.
+  // Three kind of values are possible:
+  // - `leader == nullptr`: This section was not eligible for ICF.
+  // - `leader == this`: This section was retained.
+  // - `leader != this`: This section was merged with another idential section.
+  //                     Implies `killed_by_icf`.
```

Comment:
Does this mean we can get rid of `killed_by_icf` flag and use this pointer instead?

Reply:
It’s assigned [here](https://github.com/rui314/mold/blob/b3ed06aa3956ed309ad2fe9ab757f6ee87f4f467/elf/icf.cc#L598), and it’s assigned from a single place, so we should be able to replace this with a helper function denoting the same condition.

Reply:
Do you mind if I ask you to do this?

Reply:
No problem, will do.